crates/protocol/src/types/len_string.rs

```diff
@@ -0,0 +1,81 @@
+//! A length-prefixed string with a length limit (prefixed by a [`super::VarInt`])
+
+use std::string;
+
+use bytes::{Buf, BufMut};
+
+use crate::{error, ProtocolType};
+
+/// Guaranteed to be no larger than `MAX_SIZE`.
+#[derive(Debug, Clone)]
+pub struct LenString<const MAX_SIZE: usize>(pub(super) String);
```

Comment:
While it is not `pub`, the inner field's invariant is too easy to break. Do we really need this `pub(super)`?

Reply:
Yes. The identifier type needs it to bypass a check that's already been done